### ISSUE TITLE
PtyProcess.spawn cwd parameter not working

### ISSUE THREAD

**Issue by Stefanhg** [andfoy/pywinpty#378]
Hello,
I found a case where the `cwd` parameter does not work. 
It is not documented how it is intended to work so therefore it can also be a misunderstanding.

If i make a folder called `my_folder `with `my_exe_file.exe` and then do:
`PtyProcess.spawn("my_exe_file", cwd=os.path.join(os.getcwd(), "my_folder))`
It fails with 
`FileNotFoundError: The command was not found or was not executable: my_exe_file.exe.`

This is because it grabs the path from `path = env.get('PATH', os.defpath)` instead of `path = cwd `
I am not sure what the idea of the `cwd` parameter is when it uses the path from `env`. 
Should it have prioritized using `cwd` and `cwd=None` it then defaults to `env.get('PATH', os.defpath)`?

**Comment by andfoy** [comment]
Hi @Stefanhg, thanks for reaching out. The `cwd` parameter is used to initialize the path from where the program will start from. That is, different from the actual location of the binary to execute, which should be available in the `PATH`. In order to execute your program properly in your example, you can either specify the full path to the binary as `C:\<long path comes here>\my_folder\my_exe_file.exe` or add `my_folder` to  `PATH`.

Please let me know if this works for you

**Comment by Stefanhg** [comment]
Hello @andfoy, 
This ended up raising more questions for me as I am really unsure how this actually work.

So the way `cwd` is implemented here is `cwd` is actually the path that will be set as working path when `executable` is executed? 
If so, then I guess It is just one of those "if documented, no problems" cases. It has just been confusing me so much as I expected `cwd` to yield the same result as adding `path_to_my_folder` to `PATH`. 

I did get it to work by adding `path_to_my_folder` to `PATH` but i still don't see how the implementation of `cwd` is supposed to be useful if you cannot use it to set `path_to_my_folder` before executing.

It is the only thing I found while using this tool. It works SO WELL for the testing I was doing! I'm glad I came across this!

**Comment by andfoy** [comment]
@Stefanhg First of all, I'm glad this library works for you! Second, here is a much lengthy explanation about the behaviour for the `cwd` argument:

According to the [CreateProcessW](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw) Windows API documentation, the resolution of an executable occurs as follows:

1. The directory from which the application loaded.
2. The current directory for the parent process.
3. The 32-bit Windows system directory. Use the GetSystemDirectory function to get the path of this directory.
4. The 16-bit Windows system directory. There is no function that obtains the path of this directory, but it is searched. The name of this directory is System.
5. The Windows directory. Use the GetWindowsDirectory function to get the path of this directory.
6. The directories that are listed in the PATH environment variable. Note that this function does not search the per-application path specified by the App Paths registry key. To include this per-application path in the search sequence, use the ShellExecute function.

For documentation purposes, the `cwd` command is being mapped internally into the `lpCurrentDirectory` argument, which specifies:

> The full path to the current directory for the process. The string can also specify a UNC path.

It is worthy to note that there is not a step in the aforementioned process where the `lpCurrentDirectory` argument is used to resolve an executable path. Thus making this library compliant with the Windows API spec.

While the confusion regarding program search paths is understandable, the current behavior is the built-in one in Windows, and given the long confusing process used to search a path described above, it is most recommended to either specify the full path to a binary or add it into the PATH.

I hope this explanation clears the confusion.

**Comment by Stefanhg** [comment]
Ah okay now I understand. Thank you for clarifying!

When i was trying to get it all working I was also initially using the full path when executing and it is also my preferred method but  It ended up not working in some way and after some time I just ditched it and tried to find alternative solutions.
Now I am looking at again I think I did resolve the issue which was mainly me forgetting to pass the string as a raw.. Typical!

I guess I can close this issue. It is just me having a different idea of how it works and a solution for my problem already exists in two ways as shown in your first comment
